saleor/graphql/channel/resolvers.py

```diff
@@ -1,8 +1,26 @@
+from typing import Optional
+
 from ...channel import models
+from ...core.permissions import is_app, is_staff_user
+from ..core.utils import from_global_id_or_error
+from ..core.validators import validate_one_of_args_is_in_query
+from .types import Channel
+
+
+def resolve_channel(info, id: Optional[str], slug: Optional[str]):
+    validate_one_of_args_is_in_query("id", id, "slug", slug)
+    if id:
+        _, db_id = from_global_id_or_error(id, Channel)
+        channel = models.Channel.objects.filter(id=db_id).first()
+    else:
+        channel = models.Channel.objects.filter(slug=slug).first()
```

Comment:
what if channel with given slug does not exists? shouldn't we raise `GraphQLError`? in this case it would return None, as well as if it gets to line: 20 `if is_staff_user(info.context) or is_app(info.context):`

Reply:
Do we raise errors when trying to fetch a product that is not public or one that does not exist?

Reply:
We do not, my bad. Got confused a bit there.